### PR TITLE
♻️ Reflect endpoints changes in handlers

### PR DESCRIPTION
**Pull request by TellFruit**

### CODE REVIEW

File: backend/Wheelingful.BLL/Models/Responses/FetchBookPaginationResponse.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Wheelingful.BLL.Models.Responses;
+
+public class FetchBookPaginationResponse
+{
+    public int PageCount { get; set; }
```
Comment (VladislavAntonyuk) [inline backend/Wheelingful.BLL/Models/Responses/FetchBookPaginationResponse.cs]: include PageSize and TotalItems in response. If user set PageSizze in query with a huge value, your db will suffer. So you Limit the PageSize and return the updated value

File: backend/Wheelingful.BLL/Models/Responses/FetchChapterPaginationResponse.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Wheelingful.BLL.Models.Responses;
+
+public class FetchChapterPaginationResponse
```
Comment (VladislavAntonyuk) [inline backend/Wheelingful.BLL/Models/Responses/FetchChapterPaginationResponse.cs]: consider making it generic for all paged responses

File: backend/Wheelingful.BLL/Services/Books/BookReaderService.cs [hunk 3]
```diff
@@ -32,7 +27,7 @@ public Task<List<FetchBookResponse>> GetBooks(FetchBookPaginationRequest request
             }
         }
 
-        return query
+        var books = await query
```
Comment (VladislavAntonyuk) [inline backend/Wheelingful.BLL/Services/Books/BookReaderService.cs]: consider creating an extension method ToPagedListAsync

File: backend/Wheelingful.BLL/Models/Responses/FetchBookPaginationResponse.cs (hunk 1 above)
Comment (TellFruit) [reply backend/Wheelingful.BLL/Models/Responses/FetchBookPaginationResponse.cs]: PageSize is already limited in validation of the request. As for totalItems - I don't quite understand the purpose.

File: backend/Wheelingful.BLL/Models/Responses/FetchChapterPaginationResponse.cs (hunk 2 above)
Comment (TellFruit) [reply backend/Wheelingful.BLL/Models/Responses/FetchChapterPaginationResponse.cs]: Good point. Will check the possibility.

File: backend/Wheelingful.BLL/Services/Books/BookReaderService.cs (hunk 3 above)
Comment (TellFruit) [reply backend/Wheelingful.BLL/Services/Books/BookReaderService.cs]: Now created.

File: backend/Wheelingful.BLL/Extensions/Generic/PaginationExtension.cs [hunk 4]
```diff
@@ -4,13 +4,14 @@ namespace Wheelingful.BLL.Extensions.Generic;
 
 public static class PaginationExtension
 {
-    public static IQueryable<T> Paginate<T>(this IQueryable<T> values, int pageNumber, int pageSize)
+    public static Task<List<T>> ToPagedListAsync<T>(this IQueryable<T> values, int pageNumber, int pageSize)
```
Comment (VladislavAntonyuk) [inline backend/Wheelingful.BLL/Extensions/Generic/PaginationExtension.cs]: I mean return Generic Response here and include all Counts

**Review by VladislavAntonyuk — APPROVED** [review]

**Pull request merged**